src/hotspot/share/cds/archiveBuilder.hpp

```diff
@@ -164,7 +166,11 @@ class ArchiveBuilder : public StackObj {
     uintx ptrmap_start()  const    { return _ptrmap_start; } // inclusive
     uintx ptrmap_end()    const    { return _ptrmap_end;   } // exclusive
     bool read_only()      const    { return _read_only;    }
+    bool has_embedded_pointer() const { return _has_embedded_pointer; }
+    void found_embedded_pointer()  { _has_embedded_pointer = true; }
```

Comment:
I'd suggest changing the function name to `set_has_embedded_pointer`.
We already have similar name (`set_has_native_pointers`) in heapShared.hpp.